spec/generators/rspec/scaffold/scaffold_generator_spec.rb

```diff
@@ -41,12 +43,13 @@
     end
 
     describe 'in an engine' do
-      before do
-        allow_any_instance_of(::Rails::Generators::NamedBase).to receive(:mountable_engine?).and_return(true)
-        run_generator %w[posts --request_specs]
+      it 'generates files with Engine url_helpers' do
+        in_sub_process do
```

Comment:
👏 

Wondering why it was wiping out the original method along the way.

Reply:
I don't know 🤔

Reply:
I suspect, and this is just a theory on my part, because when it stubs the method doesn't exist, it's then loaded, and the then stub blows it away as it was never there originally.

Reply:
![](https://64.media.tumblr.com/2d558e848ad0d16ddb97c8bf79f5b86f/tumblr_nt93mqA9ZU1qdrcyyo4_250.gif)

Reply:
:exploding_head: :scream:
Reminds me of https://github.com/rspec/rspec-rails/issues/2394#issuecomment-706372093